zeroconf/__init__.py

```diff
@@ -2576,31 +2576,44 @@ def register_service(
         ttl: Optional[int] = None,
         allow_name_change: bool = False,
         cooperating_responders: bool = False,
+        broadcast_service: bool = True,
```

Comment:
There is a use case to do this in sync code as well since you may want to register many services and then do all the broadcasts similar to how `unregister_all_services` works.

Reply:
I'm wondering if having a `register_services()` (plural) wouldn't be a better API to make this possible (and could replace `register_service()` (singular) in the long run.

What would the ideal API to cover the use case described in your comment look like, from your point of view?

Reply:
That would solve it if we could group the registrations, however in Home Assistant we end up with many integrations that have their own code bases doing the registrations so we can't group them.

Reply:
Example uses:

https://github.com/home-assistant/core/blob/dev/homeassistant/components/zeroconf/__init__.py#L247

https://github.com/ikalchev/HAP-python/blob/dev/pyhap/accessory_driver.py#L307

Reply:
All right, I'll need to think about this some more but it makes sense to have something like that.

Reply:
Now that its a `composition`, I think we can drop the legacy `ttl` arg during service registration and make this simpler.  I'll refactor this a bit later today

Reply:
After sleeping this, I think there is a bit of YAGNI going on here, and I stripped this down to only the functionality that is needed at this time.